relay-metrics/src/aggregatorservice.rs

```diff
@@ -326,6 +334,11 @@ impl AggregatorService {
         } = msg;
         self.aggregator
             .merge_all(project_key, buckets, self.max_total_bucket_bytes);
+
+        // Automatic force flush for tests:
+        if !self.aggregate {
+            self.try_flush(true);
+        }
```

Comment:
```suggestion
        // Automatic force flush for tests:
        if self.aggregate {
            self.aggregator
                .merge_all(project_key, buckets, self.max_total_bucket_bytes);
        } else if let Some(receiver) = self.receiver {
            receiver.send(FlushBuckets { partition_key: 0, buckets: HashMap::from(&[(project_key, buckets)] });
        }
```

We should be able to bypass the entire aggregator here.

Reply:
The reason I'm still merging the bucket is that `merge` calls `validate_bucket_key`, which I still want to run in tests:

https://github.com/getsentry/relay/blob/90221be076f48f37a1345a1ec7bc83af12e9f930/relay-metrics/src/aggregator.rs#L719

The alternative would be to pull out the validation into the aggregator service.

I will add a code comment to clarify.